### ISSUE TITLE
Chrome browser pod runs but no web browser running in the pod

### ISSUE THREAD

**Issue by OleksandrHerasymenko** [aerokube/moon#256]
**Versions used**
Versions affected (tested): 1.7.1, 1.8.1
Pod Chrome browser version: 86-88 (others not tested)

**Description**
Oftentimes when I try to run a browser pod (manually or automatically) via Capabilities -> Create Session the pod gets created but no web  browser inside.

**Troubleshooting details**
Case when pod is running but no browser inside:
- Right after creation the following request http://mydomain.test/wd/hub/session fails with 502 error
- Moon container log output shows the following error message
`2021/02/17 14:07:59 [PROXY_ERROR] [unknown] [10.203.17.129, 127.0.0.1] [POST http://chrome-86-0-be596045-42da-4d13-834f-5bde6b17c89f.browsers:4545/session] [read tcp 10.203.18.130:34774->10.203.17.38:4545: read: connection reset by peer]`
- Defender container log output:
`2021/02/17 14:04:49 Unsolicited response received on idle HTTP channel starting with "{\"value\":{\"error\":\"unknown command\",\"message\":\"unknown command: unknown command: \",\"stacktrace\":\"#0 0x559f0c303d99 \\u003Cunknown>\\n\"}}"; err=<nil>`

Case if pod is running and browser is up inside as well:
-  http://mydomain.test/wd/hub/session replies with 200 right after creation
- Moon container doesn't have the error mentioned above
- Defender container log output:
```
2021/02/17 14:06:01 Unsolicited response received on idle HTTP channel starting with "{\"value\":{\"error\":\"unknown command\",\"message\":\"unknown command: unknown command: \",\"stacktrace\":\"#0 0x560a5f6aed99 \\u003Cunknown>\\n\"}}"; err=<nil>
2021/02/17 14:06:02 [PROXYING] [POST http://localhost:4444/session]
```

**Summary:**
I haven't found any dependencies on the Moon version or Chrome version used. To me, the issue reproduces sporadically. More often than not it doesn't work

Thanks, your help is much appreciated.

**Comment by vania-pooh** [comment]
@OleksandrHerasymenko are you using spot instances in Kubernetes cluster?

**Comment by OleksandrHerasymenko** [comment]
Hi Vania,

we are running Moon on GKE on a standard VM types.

**Comment by aandryashin** [comment]
Hello, looks like OOM, could you please try to increase memory resourses
for browser container?

ср, 17 февр. 2021 г., 18:50 OleksandrHerasymenko <notifications@github.com>:

**Comment by OleksandrHerasymenko** [comment]
Hi @aandryashin ,

Thank you for the suggestion. We made some tests with the following config:
```
"resources": {
              "limits": {
                "memory": "2Gi"
              },
              "requests": {
                "memory": "1Gi"
              }
```
However, this didn't help. The issue still persist

**Comment by aandryashin** [comment]
Every session is failed to start or some?

чт, 18 февр. 2021 г., 12:42 OleksandrHerasymenko <notifications@github.com>:

**Comment by OleksandrHerasymenko** [comment]
Hi,

we've discovered that the issue was caused by Istio sidecar which is added to the browser pod. We assume that the Istio sidecar takes longer to start than browser container.
Thus until Istio sidecar is up and running, the network connection to the browser pod fails. We expect that [this](https://istio.io/latest/news/releases/1.8.x/announcing-1.8/change-notes/) Istio version would help us to resolve the issue by providing the `holdApplicationUntilProxyStarts ` run option. As of now, we don't have the mentioned Istio version configured so we can't tell for sure. The one thing that we currently know for sure is that Moon works fine without Istio.
Given the above, that would be great to know if Moon has some timeout setting that can help to deal with this Istio sidecar lag.
Anyways, the issue is not in the Moon field, so I'm closing this request. Thank you for your help.
